packages/compartment-mapper/src/import-hook.js

```diff
@@ -338,6 +339,20 @@ function* chooseModuleDescriptor(
     // name, they are usable as URL's.
     const moduleLocation = resolveLocation(candidateSpecifier, packageLocation);
 
+    // check if moduleLocation is within the compartment if it is not, then
+    // create a redirect record using absolute path which will be resolved by
+    // findRedirect
+    const { fileURLToPath } = unpackReadPowers(readPowers);
```

Comment:
We could avoid doing this work for every importHook invocation by moving it up to the maker.